src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -70,13 +70,18 @@ public GitToolChooser(String remoteName, Item projectContext, String credentials
         currentNode = n;
         this.listener = listener;
         implementation = "NONE";
-        useCache = decideAndUseCache(remoteName);
 
-        if (useCache) {
-            implementation = determineSwitchOnSize(sizeOfRepo, gitExe);
-        } else {
-            decideAndUseAPI(remoteName, projectContext, credentialsId, gitExe);
+        // Skip expensive checks if there is only one available implementation to choose from
+        if (JGIT_SUPPORTED) {
```

Comment:
I like this change very much, but I don't think it will address the concern you expressed in the pull request.  When I ran this pull request with the command:
```
$ mvn clean -Djenkins.version=2.263 install hpi:run
```

I confirmed that JGIT_SUPPORTED is true even if Jenkins has not enabled the JGit implementation.  The JGIT_SUPPORTED variable is used to decide if features have been requested that are not implemented in JGit.  It is not currently being used to decide that JGit is not enabled.

I think it is a good idea to skip the evaluation decision to use JGit or not use JGit if JGit is not enabled.  However, that's not what this pull request is doing at the current time.

Did this change resolve the "awful performance" you saw with git plugin performance improvement enabled in your environment?

Reply:
> I confirmed that JGIT_SUPPORTED is true even if Jenkins has not enabled the JGit implementation.

Whoops. Note to myself. Never ever trust documentation:

```
@param useJGit if true the JGit is allowed as an implementation
```

I'll search for another way to test whether JGit is enabled then.